src/user_service/validators.py

```diff
@@ -0,0 +1,309 @@
+
+        Args:
+            role (object): The role object to validate.
+
+        Raises:
+            ValidationError: If the role does not meet the expected constraints.
+        """
+
+        from .models import User
+        try:
+            role_field = User._meta.get_field("role")
+        except FieldDoesNotExist:
+            raise ValidationError(
+                _("Role field is not defined on the User model.")
+            )
+
+        if role is None and not role_field.null:
+            raise ValidationError(
+                _("Role is required for this user"),
+                code="role_required"
+            )
+
+        role_model = role_field.related_model
+        if role and not isinstance(role, role_model):
+            raise ValidationError(
+                _("Invalid role type. Expected %(expected)s, got %(actual)s"),
+                params={
+                    "expected": role_model.__name__,
+                    "actual": type(role).__name__
+                },
+                code="invalid_role_type"
+            )
+
+        if role is not None:
+            if role_field.remote_field.on_delete == models.SET_NULL and role is None:
+                return
+
+            if not hasattr(role, "name") or not role.name:
+                raise ValidationError(
+                    _("Role must have a valid name"),
+                    code="role_name_required"
+                )
+
+
+class OrgModelValidator:
+    """
+    A class containing static methods to validate organisation-related attributes.
+    """
+
+    @staticmethod
+    def name_validator(name, max_length=125):
+        """
+        Validates an organisation name based on length and allowed characters.
+
+        Args:
+            name (str): The name of the organisation.
+            max_length (int): Maximum allowed length for the name.
+
+        Raises:
+            ValidationError: If the name exceeds the length or contains invalid characters.
+        """
+        if len(name) > max_length:
+            raise ValidationError(
+                _("Organisation name cannot exceed %(max_length)d characters.") % {"max_length": max_length}
+            )
+
+        if not re.match(r'^[a-zA-Z0-9\s]+$', name):
+            raise ValidationError(
+                _("Organisation name can only contain letters, numbers and spaces.")
+            )
+
+    @staticmethod
+    def owner_validator(owner):
+        """
+        Validates the owner attribute of an organisation.
+
+        Args:
+            owner (object): The owner to be validated.
+
+        Raises:
+            ValidationError:
+                - If the owner field is not defined in the Organisation model.
+                - If the owner is not an instance of the related User model.
+        """
+        from .models import Organisation
+
+        try:
+            owner_field = Organisation._meta.get_field("owner")
+        except FieldDoesNotExist:
```

Comment:
When you catch an exception and raise another one, you should be explicit about it. You can for example do something like
```python
try:
    something_that_may_raise_an_exception
except ThOriginalException as e:
    raise SomeOtherExceptionInstead from e

Reply:
Well, exception chaining is good. Why not? Will help with logging... we can see the full chain of exceptions in the logs when debugging. 

I will make the change